File: 1669.merge-in-between-linked-lists.py
#
# @lc app=leetcode id=1669 lang=python
#
# [1669] Merge In Between Linked Lists
#

# @lc code=start
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def mergeInBetween(self, list1, a, b, list2):
        """
        :type list1: ListNode
        :type a: int
        :type b: int
        :type list2: ListNode
        :rtype: ListNode
        """
        curr = list1
        head = curr
        for _ in range(1,a):
            curr = curr.next
        remover = curr
        for _ in range(-1, b - a):
            remover = remover.next
        temp = remover
        remover = remover.next
        temp.next = None
        temp = None
        curr.next = list2
        while (curr.next is not None):
            curr = curr.next
        curr.next = remover
        return head
# @lc code=end

File: 1669.merge-in-between-linked-lists.java
/*
 * @lc app=leetcode id=1669 lang=java
 *
 * [1669] Merge In Between Linked Lists
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeInBetween(ListNode list1, int a, int b, ListNode list2) {
        ListNode curr = list1;
        ListNode head = curr;
        for (int i = 1; i < a; i++) {
            curr = curr.next;
        }
        ListNode remover = curr;
        for (int i = -1; i < b - a; i++) {
            remover = remover.next;
        }
        ListNode temp = remover;
        remover = remover.next;
        temp.next = null;
        temp = null;
        curr.next = list2;
        while (curr.next != null) {
            curr = curr.next;
        }
        curr.next = remover;
        return head;
    }
}
// @lc code=end

